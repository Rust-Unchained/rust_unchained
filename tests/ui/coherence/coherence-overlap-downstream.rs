// Tests that we consider `T: Sugar + Fruit` to be ambiguous, even
// though no impls are found.

pub trait Sugar {}
pub trait Fruit {}
pub trait Sweet {}

// UNCHAINED_TODO: This should be allowed in the future,
// conflicts should only be reported if they do happen.
impl<T:Sugar> Sweet for T { }
impl<T:Fruit> Sweet for T { }
//~^ ERROR E0119

pub trait Foo<X> {}
pub trait Bar<X> {}
impl<X, T> Foo<X> for T where T: Bar<X> {}
// UNCHAINED_TODO: This causes conflicts in standard Rust, but not in unchained, for some reason.
// Although this is desired behavior since i32 does not implement `Bar`, I left this `TODO` so that this maybe reviewed later.
impl<X> Foo<X> for i32 {}

fn main() { }
