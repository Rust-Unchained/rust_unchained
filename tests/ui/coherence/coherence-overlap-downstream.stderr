error[E0119]: conflicting implementations of trait `Sweet`
  --> $DIR/coherence-overlap-downstream.rs:11:1
   |
LL | impl<T:Sugar> Sweet for T { }
   | ------------------------- first implementation here
LL | impl<T:Fruit> Sweet for T { }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0119`.
